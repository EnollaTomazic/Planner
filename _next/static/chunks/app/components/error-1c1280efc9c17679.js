(self.webpackChunk_N_E=self.webpackChunk_N_E||[]).push([[872],{50960:(e,s,r)=>{"use strict";r.d(s,{A:()=>o,R:()=>c});var a=r(95155),n=r(12115),i=r(52619),t=r.n(i),l=r(24625);function c(e){let{error:s,reset:r,title:i,description:c,retryLabel:o="Try again",homeLabel:d="Go to dashboard",homeHref:h="/"}=e;return(0,n.useEffect)(()=>{console.error(s)},[s]),(0,a.jsx)(l.qK,{as:"section",role:"alert","aria-live":"assertive",className:"py-[var(--space-8)]",children:(0,a.jsxs)("div",{className:"flex max-w-prose flex-col gap-[var(--space-4)]",children:[(0,a.jsxs)("div",{className:"space-y-[var(--space-2)]",children:[(0,a.jsx)("h1",{className:"text-display-sm font-semibold text-foreground",children:i}),(0,a.jsx)("p",{className:"text-body text-muted-foreground",children:c})]}),(0,a.jsxs)("div",{className:"flex flex-wrap gap-[var(--space-3)]",children:[(0,a.jsx)(l.$n,{variant:"primary",onClick:r,children:o}),(0,a.jsx)(l.$n,{asChild:!0,variant:"ghost",children:(0,a.jsx)(t(),{href:h,children:d})})]})]})})}function o(e){return(0,a.jsx)(c,{...e,title:"Something went wrong",description:"This section hit an error, but the rest of Planner is still running. Try again or return home."})}},59729:(e,s,r)=>{"use strict";r.r(s),r.d(s,{default:()=>a.A});var a=r(50960)},94822:(e,s,r)=>{Promise.resolve().then(r.bind(r,59729))}},e=>{e.O(0,[6428,9295,2619,4625,8441,1255,7358],()=>e(e.s=94822)),_N_E=e.O()}]);