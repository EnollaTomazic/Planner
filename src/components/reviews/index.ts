// src/components/reviews/index.ts
export { default as ReviewPage } from "./ReviewPage";
export { default as ReviewsPage } from "./ReviewsPage";
export { default as ReviewEditor } from "./ReviewEditor";
export { default as ReviewList } from "./ReviewList";
export { default as ReviewListItem } from "./ReviewListItem";
export { default as ReviewSummary } from "./ReviewSummary";
export { default as ReviewSummaryHeader } from "./ReviewSummaryHeader";
export { default as ReviewSummaryScore } from "./ReviewSummaryScore";
export { default as ScoreMeter } from "./ScoreMeter";
export { default as ReviewSummaryPillars } from "./ReviewSummaryPillars";
export { default as ReviewSummaryTimestamps } from "./ReviewSummaryTimestamps";
export { default as ReviewSummaryNotes } from "./ReviewSummaryNotes";
export { default as RoleSelector } from "./RoleSelector";
export { default as ReviewPanel } from "./ReviewPanel";
export { default as NeonIcon } from "./NeonIcon";
export { useReviewFilter } from "./useReviewFilter";
export { default as LaneOpponentForm } from "./LaneOpponentForm";
export { default as ResultScoreSection } from "./ResultScoreSection";
export { default as PillarsSelector } from "./PillarsSelector";
export { default as TimestampMarkers } from "./TimestampMarkers";
