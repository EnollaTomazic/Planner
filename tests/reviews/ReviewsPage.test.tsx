import React from "react";
import {
  render,
  screen,
  fireEvent,
  cleanup,
  within,
} from "@testing-library/react";
import { describe, it, expect, afterEach, vi } from "vitest";
import { ReviewsPage } from "@/components/reviews";
import type { Review } from "@/lib/types";

afterEach(() => {
  cleanup();
  vi.useRealTimers();
});

const baseReviews: Review[] = [
  {
    id: "1",
    title: "Alpha",
    tags: [],
    pillars: [],
    createdAt: 3000,
    matchup: "Lux vs Ahri",
    role: "MID",
  },
  {
    id: "2",
    title: "Beta",
    tags: [],
    pillars: [],
    createdAt: 1000,
    matchup: "Garen vs Darius",
    role: "TOP",
  },
  {
    id: "3",
    title: "Gamma",
    tags: [],
    pillars: [],
    createdAt: 2000,
    matchup: "Ashe vs Cait",
    role: "BOT",
  },
];

describe("ReviewsPage", () => {
  it("renders default state", () => {
    const { container } = render(
      <ReviewsPage
        reviews={baseReviews}
        selectedId={null}
        onSelect={() => {}}
        onCreate={() => {}}
        onRename={() => {}}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it("filters reviews by search query", async () => {
    render(
      <ReviewsPage
        reviews={baseReviews}
        selectedId={null}
        onSelect={() => {}}
        onCreate={() => {}}
        onRename={() => {}}
      />,
    );

    const search = screen.getByRole("searchbox");
    vi.useFakeTimers();
    fireEvent.change(search, { target: { value: "Gamma" } });
    vi.advanceTimersByTime(250);
    await Promise.resolve();
    expect(screen.getByText("Gamma")).toBeInTheDocument();
    expect(screen.queryByText("Alpha")).toBeNull();
    expect(screen.queryByText("Beta")).toBeNull();
    vi.useRealTimers();
  });

  it("sorts reviews by title", () => {
    render(
      <ReviewsPage
        reviews={baseReviews}
        selectedId={null}
        onSelect={() => {}}
        onCreate={() => {}}
        onRename={() => {}}
      />,
    );

    const list = screen.getAllByRole("list")[0];
    const getTitles = () =>
      within(list)
        .getAllByRole("button")
        .map((b) => b.getAttribute("aria-label")?.replace("Open review: ", ""));

    expect(getTitles()).toEqual(["Alpha", "Gamma", "Beta"]);

    const sortBtn = screen.getByRole("button", { name: "Sort reviews" });
    fireEvent.click(sortBtn);
    fireEvent.click(screen.getByRole("option", { name: "Title" }));

    expect(getTitles()).toEqual(["Alpha", "Beta", "Gamma"]);
  });

  it("creates a new review and opens editor", () => {
    function Harness({ onCreate }: { onCreate: () => void }) {
      const [reviews, setReviews] = React.useState<Review[]>(baseReviews);
      const [selectedId, setSelectedId] = React.useState<string | null>(null);
      return (
        <ReviewsPage
          reviews={reviews}
          selectedId={selectedId}
          onSelect={setSelectedId}
          onCreate={() => {
            onCreate();
            const next: Review = {
              id: "4",
              title: "New Review",
              tags: [],
              pillars: [],
              createdAt: 4000,
            };
            setReviews((r) => [...r, next]);
            setSelectedId(next.id);
          }}
          onRename={() => {}}
        />
      );
    }

    const createSpy = vi.fn();
    render(<Harness onCreate={createSpy} />);

    fireEvent.click(screen.getByRole("button", { name: "New Review" }));
    expect(createSpy).toHaveBeenCalled();
    expect(screen.getByRole("button", { name: "Done" })).toBeInTheDocument();
  });

  it("toggles between summary and edit panels", () => {
    function Harness() {
      const [selectedId, setSelectedId] = React.useState<string>("1");
      return (
        <ReviewsPage
          reviews={baseReviews}
          selectedId={selectedId}
          onSelect={setSelectedId}
          onCreate={() => {}}
          onRename={() => {}}
        />
      );
    }

    render(<Harness />);

    fireEvent.click(screen.getByRole("button", { name: "Edit review" }));
    expect(screen.getByRole("button", { name: "Done" })).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button", { name: "Done" }));
    expect(screen.getByRole("button", { name: "Edit review" })).toBeInTheDocument();
  });
});
