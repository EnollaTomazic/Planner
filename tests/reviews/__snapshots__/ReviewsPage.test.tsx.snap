// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReviewsPage > renders default state 1`] = `
<div>
  <main
    aria-labelledby="reviews-header"
    class="page-shell py-6 space-y-6"
  >
    <section
      class="sticky top-0"
    >
      <style
        global="true"
        jsx="true"
      >
        
      .hero2-neomorph {
        background: linear-gradient(
          145deg,
          hsl(var(--card)),
          hsl(var(--panel))
        );
        --hero2-shadow-key: var(--shadow);
        --hero2-shadow-ambient: inset 0 0 0 calc(var(--hairline-w) * 1.5)
          hsl(var(--shadow-color) / 0.18);
        box-shadow: var(--hero2-shadow-ambient), var(--hero2-shadow-key);
      }
      @supports (color: color-mix(in oklab, white, black)) {
        .hero2-neomorph {
          --hero2-shadow-ambient: inset 0 0 0 calc(var(--hairline-w) * 1.5)
            color-mix(
              in oklab,
              hsl(var(--shadow-color)) 28%,
              hsl(var(--highlight))
            );
        }
      }
      @media (prefers-contrast: more) {
        .hero2-frame {
          border-color: hsl(var(--foreground) / 0.7) !important;
        }
        .hero2-neomorph {
          --hero2-shadow-key: 0 0 var(--space-4)
            hsl(var(--foreground) / 0.75);
          --hero2-shadow-ambient: inset 0 0 0 calc(var(--hairline-w) * 2)
            hsl(var(--foreground) / 0.7);
          box-shadow: var(--hero2-shadow-ambient), var(--hero2-shadow-key);
        }
      }
      @media (forced-colors: active) {
        .hero2-frame {
          border-color: CanvasText !important;
          background: Canvas !important;
        }
        .hero2-neomorph {
          background: Canvas !important;
          box-shadow: none !important;
        }
      }
    
      </style>
      <div
        class="relative overflow-visible hero2-frame hero2-neomorph rounded-card r-card-lg border border-border/40 bg-card/70 shadow-outline-subtle md:px-[var(--space-7)] md:py-[var(--space-7)] lg:px-[var(--space-8)] lg:py-[var(--space-8)] rounded-card r-card-lg px-4 py-4"
      >
        <div
          class="relative z-[2] space-y-[var(--space-5)] md:space-y-[var(--space-6)]"
        >
          <div
            class="relative z-[2] space-y-2"
          >
            <header
              class="z-[999] relative isolate after:absolute after:left-0 after:bottom-0 after:h-px after:w-full after:bg-gradient-to-r after:from-primary after:via-accent after:to-transparent after:z-[2]"
              id="reviews-header"
            >
              <div
                class="sticky top-[var(--header-stack)] relative flex items-center gap-[var(--space-3)] sm:gap-[var(--space-4)] px-[var(--space-3)] sm:px-[var(--space-4)] py-[var(--space-3)] sm:py-[var(--space-4)] min-h-[var(--space-7)]"
              >
                <div
                  aria-hidden="true"
                  class="header-rail pointer-events-none absolute left-0 top-[var(--space-1)] bottom-[var(--space-1)] w-[var(--space-2)] rounded-l-2xl"
                />
                <div
                  class="flex min-w-0 flex-1 items-center gap-[var(--space-3)] sm:gap-[var(--space-4)]"
                >
                  <div
                    class="flex min-w-0 items-center gap-[var(--space-2)] sm:gap-[var(--space-3)]"
                  >
                    <span
                      class="shrink-0 opacity-90"
                    >
                      <svg
                        aria-hidden="true"
                        class="lucide lucide-book-open opacity-80"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 7v14"
                        />
                        <path
                          d="M3 18a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h5a4 4 0 0 1 4 4 4 4 0 0 1 4-4h5a1 1 0 0 1 1 1v13a1 1 0 0 1-1 1h-6a3 3 0 0 0-3 3 3 3 0 0 0-3-3z"
                        />
                      </svg>
                    </span>
                    <div
                      class="min-w-0"
                    >
                      <div
                        class="flex min-w-0 items-baseline gap-[var(--space-2)]"
                      >
                        <h1
                          class="text-balance break-words text-title leading-tight text-foreground sm:text-title-lg font-semibold tracking-[-0.01em]"
                        >
                          Reviews
                        </h1>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </header>
            <section>
              <div
                class="sticky sticky-blur top-[var(--header-stack)]"
              >
                <div
                  class="relative z-[2] flex items-center gap-[var(--space-2)] md:gap-[var(--space-3)] lg:gap-[var(--space-4)] py-[var(--space-4)]"
                >
                  <span
                    aria-hidden="true"
                    class="rail"
                  />
                  <div
                    class="min-w-0"
                  >
                    <div
                      class="flex min-w-0 flex-wrap items-baseline gap-x-[var(--space-2)] gap-y-[var(--space-1)]"
                    >
                      <h2
                        class="font-semibold tracking-[-0.01em] text-balance break-words text-foreground text-title md:text-title"
                        data-text="Browse Reviews"
                      >
                        Browse Reviews
                      </h2>
                      <span
                        class="text-ui md:text-body text-muted-foreground break-words font-normal"
                      >
                        <span
                          class="pill"
                        >
                          Total 
                          3
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </section>
          </div>
        </div>
        <div
          class="relative z-[2] mt-[var(--space-6)] md:mt-[var(--space-7)] border-t border-border/35 pt-[var(--space-5)] md:pt-[var(--space-6)] grid md:grid-cols-12 md:gap-[var(--space-4)] gap-[var(--space-4)]"
          role="group"
        >
          <div
            class="flex flex-col gap-[var(--space-2)] md:col-span-7 md:justify-self-stretch"
            data-area="search"
          >
            <form
              class="grid grid-cols-[minmax(0,1fr)_auto] items-center gap-[var(--space-2)] data-[loading=true]:opacity-[var(--loading)] data-[loading=true]:pointer-events-none w-full max-w-[calc(var(--space-8)*10)] rounded-full flex-1"
              role="search"
            >
              <div
                class="relative min-w-0"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-search pointer-events-none absolute left-[var(--space-4)] top-1/2 -translate-y-1/2 size-[var(--space-4)] text-muted-foreground"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m21 21-4.34-4.34"
                  />
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                </svg>
                <div
                  class="relative inline-flex items-center border bg-card/60 backdrop-blur-[2px] shadow-control transition-[box-shadow,transform] duration-[var(--dur-quick)] ease-out motion-reduce:transition-none hover:bg-[--hover] active:bg-[--active] hover:border-[--border-hover] active:border-[--border-active] [--border-hover:hsl(var(--border)/0.38)] [--border-active:hsl(var(--border)/0.5)] hover:shadow-control-hover focus-within:outline-none focus-within:ring-2 focus-within:ring-[hsl(var(--ring))] data-[loading=true]:opacity-[var(--loading)] before:pointer-events-none before:absolute before:inset-0 before:rounded-[inherit] before:opacity-[0.05] before:bg-[repeating-linear-gradient(0deg,hsl(var(--accent-2)/0.4)_0_1px,transparent_1px_3px),var(--asset-noise-url,none)] after:pointer-events-none after:absolute after:inset-0 after:rounded-[inherit] after:p-px after:opacity-0 after:transition-opacity after:duration-[var(--dur-quick)] after:ease-out after:motion-reduce:transition-none after:bg-[var(--edge-iris,var(--accent))] after:[mask:linear-gradient(hsl(var(--foreground)),hsl(var(--foreground)))_content-box,linear-gradient(hsl(var(--foreground)),hsl(var(--foreground)))] after:[mask-composite:exclude] focus-within:after:opacity-100 w-full hero2-neomorph !border border-[hsl(var(--border)/0.4)] !shadow-neo-inset hover:!shadow-neo-soft active:!shadow-neo-inset focus-within:!shadow-neo-soft [--hover:transparent] [--active:transparent] rounded-full [&>input]:rounded-full overflow-hidden hero2-frame"
                  style="--control-h: var(--control-h-md);"
                >
                  <input
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class="w-full rounded-[inherit] bg-transparent px-[var(--space-3)] text-ui text-foreground placeholder:text-muted-foreground/70 caret-accent border-none focus:outline-none focus-visible:outline-none h-[var(--control-h)] hover:bg-[--hover] active:bg-[--active] disabled:opacity-[var(--disabled)] disabled:cursor-not-allowed read-only:cursor-default data-[loading=true]:opacity-[var(--loading)] pl-[var(--space-7)]"
                    id=":r1:"
                    name=":r1:"
                    placeholder="Search title, tags, opponent, patch…"
                    spellcheck="false"
                    type="search"
                    value=""
                  />
                </div>
              </div>
            </form>
          </div>
          <div
            class="flex flex-wrap items-center justify-end gap-[var(--space-2)] md:col-span-5 md:justify-self-end"
            data-area="actions"
          >
            <div
              class="flex flex-col gap-[var(--space-2)] sm:flex-row sm:items-center sm:gap-[var(--space-3)]"
            >
              <label
                class="flex w-full flex-col gap-[var(--space-1)] sm:w-auto sm:flex-row sm:items-center sm:gap-[var(--space-2)]"
              >
                <span
                  class="text-label font-medium text-muted-foreground"
                >
                  Sort
                </span>
                <div
                  class="glitch-wrap w-full sm:w-auto"
                >
                  <div
                    class="group inline-flex rounded-[var(--radius-full)] border border-[--theme-ring] focus-within:ring-2 focus-within:ring-[--theme-ring] focus-within:ring-offset-0"
                  >
                    <button
                      aria-controls=":r2:-listbox"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-label="Sort reviews"
                      class="_glitchTrigger_843152 relative flex items-center h-[var(--control-h-sm)] rounded-[var(--radius-full)] px-[var(--space-3)] overflow-hidden bg-muted/12 hover:bg-muted/18 focus:[outline:none] focus-visible:[outline:none] transition-colors duration-[var(--dur-quick)] ease-out motion-reduce:transition-none !h-[var(--control-h-lg)] !px-[var(--space-4)]"
                      data-lit="true"
                      data-open="false"
                      type="button"
                    >
                      <span
                        class="font-medium _glitchText_843152 text-foreground group-hover:text-foreground"
                      >
                        Newest
                      </span>
                      <svg
                        aria-hidden="true"
                        class="lucide lucide-chevron-down _caret_843152 ml-auto size-[var(--space-4)] shrink-0 opacity-75"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="m6 9 6 6 6-6"
                        />
                      </svg>
                      <span
                        aria-hidden="true"
                        class="_gbIris_843152"
                      />
                      <span
                        aria-hidden="true"
                        class="_gbChroma_843152"
                      />
                      <span
                        aria-hidden="true"
                        class="_gbFlicker_843152"
                      />
                      <span
                        aria-hidden="true"
                        class="_gbScan_843152"
                      />
                    </button>
                  </div>
                </div>
              </label>
              <button
                class="relative inline-flex items-center justify-center rounded-[var(--control-radius)] border font-medium tracking-[0.02em] transition-all duration-[var(--dur-quick)] ease-out motion-reduce:transition-none hover:bg-[--hover] active:bg-[--active] focus-visible:[outline:none] focus-visible:ring-2 focus-visible:ring-[--focus] disabled:opacity-[var(--disabled)] disabled:pointer-events-none data-[loading=true]:opacity-[var(--loading)] data-[disabled=true]:opacity-[var(--disabled)] data-[disabled=true]:pointer-events-none h-[var(--control-h-lg)] text-title gap-[var(--space-4)] [&_svg]:size-[var(--space-8)] w-full whitespace-nowrap px-[var(--space-4)] sm:w-auto shadow-glow-sm hover:shadow-glow-md active:translate-y-px active:shadow-btn-primary-active bg-[hsl(var(--primary)/0.12)] border-[hsl(var(--primary)/0.35)] text-[hsl(var(--primary-foreground))] [--hover:theme('colors.interaction.primary.hover')] [--active:theme('colors.interaction.primary.active')]"
                style="--glow-active: hsl(var(--primary) / 0.35); --btn-primary-hover-shadow: 0 2px 6px -1px hsl(var(--primary) / 0.25); --btn-primary-active-shadow: inset 0 0 0 1px hsl(var(--primary) / 0.6);"
                tabindex="0"
                type="button"
              >
                <span
                  class="absolute inset-0 pointer-events-none rounded-[inherit] bg-[linear-gradient(90deg,hsl(var(--primary)/.18),hsl(var(--primary)/.18))]"
                />
                <span
                  class="relative z-10 inline-flex items-center gap-[var(--space-2)]"
                >
                  <svg
                    aria-hidden="true"
                    class="lucide lucide-plus"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5 12h14"
                    />
                    <path
                      d="M12 5v14"
                    />
                  </svg>
                  <span>
                    New Review
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="absolute inset-0 rounded-[inherit] ring-1 ring-inset ring-border/55"
        />
      </div>
    </section>
    <div
      class="grid grid-cols-1 items-start gap-6 md:grid-cols-12"
    >
      <nav
        aria-label="Review list"
        class="md:col-span-4"
      >
        <div
          class="card-neo-soft rounded-card r-card-lg overflow-hidden bg-card/50 shadow-neo-strong"
        >
          <div
            class="section-b"
          >
            <div
              class="mb-2 text-ui text-muted-foreground"
            >
              3
               shown
            </div>
            <div
              class="rounded-card r-card-lg border border-border/25 bg-card/60 shadow-outline-subtle w-full mx-auto backdrop-blur-sm h-[calc(100vh-var(--header-stack)-var(--space-6))] overflow-auto p-2"
            >
              <ul
                class="flex flex-col gap-3"
                role="listbox"
              >
                <li>
                  <button
                    aria-label="Open review: Alpha"
                    aria-selected="false"
                    class="relative w-full text-left rounded-card r-card-lg p-3 bg-card/90 border border-border/35 transition-all duration-200 focus-visible:outline-none disabled:opacity-60 disabled:pointer-events-none disabled:bg-muted/20 hover:bg-accent/10 hover:ring-2 hover:ring-theme focus-visible:bg-accent/15 focus-visible:ring-2 focus-visible:ring-theme active:bg-accent/20 active:ring-2 active:ring-theme data-[selected=true]:bg-accent/20 data-[selected=true]:ring-2 data-[selected=true]:ring-accent"
                    data-scope="reviews"
                    role="option"
                    type="button"
                  >
                    <div
                      class="flex flex-col gap-2"
                    >
                      <div
                        class="flex items-start justify-between gap-2"
                      >
                        <div
                          class="truncate font-medium text-body"
                        >
                          Alpha
                        </div>
                        <div
                          class="truncate text-ui text-muted-foreground"
                        >
                          Lux vs Ahri
                        </div>
                      </div>
                      <div
                        class="flex items-center justify-between gap-2"
                      >
                        <div
                          class="flex items-center gap-2"
                        >
                          <span
                            aria-hidden="true"
                            class="h-2 w-2 rounded-full ring-2 motion-safe:animate-[blink_1s_steps(2)_infinite] motion-reduce:animate-none bg-muted-foreground ring-muted-foreground"
                          />
                          <span
                            class="inline-flex max-w-full items-center gap-[var(--space-2)] whitespace-nowrap rounded-card r-card-lg font-medium tracking-[-0.01em] border bg-muted/18 shadow-badge transition-[background,box-shadow,transform] duration-140 ease-out text-label leading-none border-card-hairline px-1 py-0"
                          >
                            MID
                          </span>
                        </div>
                      </div>
                    </div>
                  </button>
                </li>
                <li>
                  <button
                    aria-label="Open review: Gamma"
                    aria-selected="false"
                    class="relative w-full text-left rounded-card r-card-lg p-3 bg-card/90 border border-border/35 transition-all duration-200 focus-visible:outline-none disabled:opacity-60 disabled:pointer-events-none disabled:bg-muted/20 hover:bg-accent/10 hover:ring-2 hover:ring-theme focus-visible:bg-accent/15 focus-visible:ring-2 focus-visible:ring-theme active:bg-accent/20 active:ring-2 active:ring-theme data-[selected=true]:bg-accent/20 data-[selected=true]:ring-2 data-[selected=true]:ring-accent"
                    data-scope="reviews"
                    role="option"
                    type="button"
                  >
                    <div
                      class="flex flex-col gap-2"
                    >
                      <div
                        class="flex items-start justify-between gap-2"
                      >
                        <div
                          class="truncate font-medium text-body"
                        >
                          Gamma
                        </div>
                        <div
                          class="truncate text-ui text-muted-foreground"
                        >
                          Ashe vs Cait
                        </div>
                      </div>
                      <div
                        class="flex items-center justify-between gap-2"
                      >
                        <div
                          class="flex items-center gap-2"
                        >
                          <span
                            aria-hidden="true"
                            class="h-2 w-2 rounded-full ring-2 motion-safe:animate-[blink_1s_steps(2)_infinite] motion-reduce:animate-none bg-muted-foreground ring-muted-foreground"
                          />
                          <span
                            class="inline-flex max-w-full items-center gap-[var(--space-2)] whitespace-nowrap rounded-card r-card-lg font-medium tracking-[-0.01em] border bg-muted/18 shadow-badge transition-[background,box-shadow,transform] duration-140 ease-out text-label leading-none border-card-hairline px-1 py-0"
                          >
                            BOT
                          </span>
                        </div>
                      </div>
                    </div>
                  </button>
                </li>
                <li>
                  <button
                    aria-label="Open review: Beta"
                    aria-selected="false"
                    class="relative w-full text-left rounded-card r-card-lg p-3 bg-card/90 border border-border/35 transition-all duration-200 focus-visible:outline-none disabled:opacity-60 disabled:pointer-events-none disabled:bg-muted/20 hover:bg-accent/10 hover:ring-2 hover:ring-theme focus-visible:bg-accent/15 focus-visible:ring-2 focus-visible:ring-theme active:bg-accent/20 active:ring-2 active:ring-theme data-[selected=true]:bg-accent/20 data-[selected=true]:ring-2 data-[selected=true]:ring-accent"
                    data-scope="reviews"
                    role="option"
                    type="button"
                  >
                    <div
                      class="flex flex-col gap-2"
                    >
                      <div
                        class="flex items-start justify-between gap-2"
                      >
                        <div
                          class="truncate font-medium text-body"
                        >
                          Beta
                        </div>
                        <div
                          class="truncate text-ui text-muted-foreground"
                        >
                          Garen vs Darius
                        </div>
                      </div>
                      <div
                        class="flex items-center justify-between gap-2"
                      >
                        <div
                          class="flex items-center gap-2"
                        >
                          <span
                            aria-hidden="true"
                            class="h-2 w-2 rounded-full ring-2 motion-safe:animate-[blink_1s_steps(2)_infinite] motion-reduce:animate-none bg-muted-foreground ring-muted-foreground"
                          />
                          <span
                            class="inline-flex max-w-full items-center gap-[var(--space-2)] whitespace-nowrap rounded-card r-card-lg font-medium tracking-[-0.01em] border bg-muted/18 shadow-badge transition-[background,box-shadow,transform] duration-140 ease-out text-label leading-none border-card-hairline px-1 py-0"
                          >
                            TOP
                          </span>
                        </div>
                      </div>
                    </div>
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </nav>
      <div
        aria-live="polite"
        class="md:col-span-8"
      >
        <div
          aria-live="polite"
          class="rounded-card r-card-lg p-[var(--space-4)] border border-border/25 bg-card/60 shadow-outline-subtle w-full container mx-auto flex flex-col items-center justify-center gap-2 py-7 text-ui text-muted-foreground"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-ghost h-6 w-6 opacity-60"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 10h.01"
            />
            <path
              d="M15 10h.01"
            />
            <path
              d="M12 2a8 8 0 0 0-8 8v12l3-3 2.5 2.5L12 19l2.5 2.5L17 19l3 3V10a8 8 0 0 0-8-8z"
            />
          </svg>
          <p>
            Select a review from the list or create a new one.
          </p>
        </div>
      </div>
    </div>
  </main>
</div>
`;
